### DISCUSSION TITLE
GnuPG Dependency

### DISCUSSION THREAD

**Discussion by pcjmfranken** [Homebrew/discussions#3524] [Everyday usage] [answered]
Homebrew's `GnuPG` depends on `GnuTLS`, but the GnuPG website lists[^1] `ntbTLS` as their current TLS library.

I'm not actually sure if these packages are interchangeable, but the current `GnuTLS` dep is kind of heavy.

[^1]: https://www.gnupg.org/download/index.html

**Comment by SMillerDev** [comment]
Feel free to make a pull request to adjust it.

**Reply by pcjmfranken** [reply]
> Feel free to make a pull request to adjust it.

Absolutely will make that PR if it makes sense to do - something I wasn't yet sure about. It looks like @carlocab has kindly provided some insight below that should help towards figuring this out.

**Comment by carlocab** [comment, marked as answer]
I'd be really wary of changing the TLS implementation used by gnupg. For a few reasons:
1. gnutls is the TLS implementation used by Debian, Fedora, and Arch.
2. ntbtls is, per the [homepage](https://gnupg.org/software/ntbtls/index.html), "NOT WELL TESTED".
3. gnutls + dependencies is only 160 MB on-disk, which is hardly heavy and not really worth doing very much about. (It would be significant for embedded systems, but that's not really `brew`'s intended use-case.)

I'm happy for someone to add an `ntbtls` formula to homebrew/core, though. We can explore using it later if it matures and gets wider use.

**Reply by pcjmfranken** [reply]
Those are fair points, thanks for pointing them out. The current setup seems to work just fine too.

Let's not swap them out then. I'll see about getting a PR for that `ntbtls` formula up sometime next week.
